tests/test_meteodatalab/test_ogd_api.py

```diff
@@ -111,23 +111,48 @@ def test_download_from_ogd(
     mock_session: mock.MagicMock, tmp_path: Path, with_headers: bool
 ):
     content = b"some content"
-    href = "https://test.com/path/to/some-file.grib"
-    body = {
-        "features": [{"assets": {"some-asset.ext": {"href": href}}}],
+    main_href = "https://test.com/path/to/some-file.grib"
+    horizontal_href = "https://test.com/path/to/horizontal.grib"
+    vertical_href = "https://test.com/path/to/vertical.grib"
+
+    mock_post_response = mock.Mock()
+    mock_post_response.json.return_value = {
+        "features": [{"assets": {"some-asset.ext": {"href": main_href}}}],
         "links": [],
     }
-    mock_post_response = mock.Mock(**{"json.return_value": body})
+    mock_post_response.raise_for_status = mock.Mock()
+
     if with_headers:
         headers = {"X-Amz-Meta-Sha256": hashlib.sha256(content).hexdigest()}
     else:
         headers = {}
-    mock_get_response = mock.Mock(
-        headers=headers, **{"iter_content.return_value": [content]}
-    )
+
+    def mock_get_response(url, *args, **kwargs):
+        # Respond with asset list for coordinate URLs
+        if "assets" in url:
```

Comment:
The content of the URL should be matched more precisely because forming the URL is part of the logic that we want to test so it would be good if the test fails when that logic fails.